tests/test_events/test_api.py

```diff
@@ -122,6 +139,48 @@ def test_send(self, mock_send: MagicMock) -> None:
         )
         assert self.event_api.get_all_sent_events() == [event]
 
+    @patch.object(Boto3Client, "eventbridge")
+    def test__send__sends_events_in_chunks_respecting_limits(self, mock_send: MagicMock) -> None:
+        for i in range(33):  # AWS allows sending maximum 10 events at once
+            self.event_api.register(MyTestEvent({"x": i}))
+
+        self.event_api.send()
+
+        assert mock_send.put_events.call_count == 4
+        assert len(mock_send.put_events.call_args_list[0].kwargs["Entries"]) == 10
+        assert len(mock_send.put_events.call_args_list[1].kwargs["Entries"]) == 10
+        assert len(mock_send.put_events.call_args_list[2].kwargs["Entries"]) == 10
+        assert len(mock_send.put_events.call_args_list[3].kwargs["Entries"]) == 3
+        assert len(self.event_api.sent_events) == 33
+        assert len(self.event_api.pending_events) == 0
+        assert len(self.event_api.failed_events) == 0
+
+    @patch.object(Boto3Client, "eventbridge")
+    def test__send__always_tries_to_send_all_events_treating_each_chunk_individually(
+        self, mock_send: MagicMock, caplog: LogCaptureFixture
+    ) -> None:
+        mock_send.put_events.side_effect = (
+            None,  # no error == success
+            ValueError("Event data is too big to be sent"),
+            None,  # no error == success
+            ValueError("Event type cannot be recognized"),
```

Comment:
_FYI:_ These errors do not matter that much but I put here message to show logs more correctly 😉